cohort-cli/src/main/java/com/ibm/cohort/cli/ParameterHelper.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * (C) Copyright IBM Corp. 2020, 2020
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package com.ibm.cohort.cli;
+
+import java.math.BigDecimal;
+import java.time.OffsetDateTime;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.opencds.cqf.cql.engine.runtime.Code;
+import org.opencds.cqf.cql.engine.runtime.DateTime;
+import org.opencds.cqf.cql.engine.runtime.Interval;
+import org.opencds.cqf.cql.engine.runtime.Quantity;
+import org.opencds.cqf.cql.engine.runtime.Time;
+
+public class ParameterHelper {
+	/**
+	 * Conversion routine for CQL parameter values encoded for command line
+	 * interaction.
+	 * 
+	 * @param arguments list of CQL parameter values encoded as strings
+	 * @return decoded parameter values formatted for consumption by the CQL engine
+	 */
+	public static Map<String, Object> parseParameters(List<String> arguments) {
+		Map<String, Object> result = new HashMap<>();
+
+		Pattern p = Pattern.compile("(?<name>[^:]+):(?<type>[^:]+):(?<value>.*)");
+		for (String arg : arguments) {
+			Matcher m = p.matcher(arg);
+			if (m.matches()) {
+				String name = m.group("name");
+				String type = m.group("type");
+				String value = m.group("value");
+
+				Object typedValue = null;
+				String[] parts = null;
+				switch (type) {
+				case "integer":
```

Comment:
Slight preference to see this as an enum, but I recognize that's not a common thing in the open engines codebase.